dockertest/dockercmd.py

```diff
@@ -398,21 +399,20 @@ def wait(self, timeout=None):
     @property
     def done(self):
         """
-        Return True if processes has ended or timeout reached
+        Return True if processes has ended (successful or not)
 
-        :raises DockerTestError: on incorrect usage
+        :raises DockerTestError: execute() was not called first
+        :raises DockerCommandError: If timeout exceeded
         """
 
         if self._async_job is None:
             raise DockerTestError("Attempted to wait for done before execute()"
                                   " called.")
-        has_exited = self._async_job.sp.poll() is not None
-        timed_out = self.duration >= self.timeout
-        # Don't assume this condition is logged elsewhere
-        if timed_out:
-            self.subtest.logwarning("Async. docker command has timed out"
-                                    " after %0.2f", float(self.timeout))
-        return has_exited or timed_out
+        if self.duration >= self.timeout:
+            raise DockerCommandError("Async. docker command has timed out"
```

Comment:
Future debuggers might appreciate a `str(self)` indicating what the command was...